integration_tests/auction_test.go

```diff
@@ -0,0 +1,239 @@
+package integration_tests
+
+import (
+	"context"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/peggyjv/sommelier/v4/x/auction/types"
+	cellarfees "github.com/peggyjv/sommelier/v4/x/cellarfees/types"
+)
+
+func (s *IntegrationTestSuite) TestAuctionModule() {
+	s.Run("Bring up chain, test governance proposal to set token prices, submit some bids, and finish an auction", func() {
+		val := s.chain.validators[0]
+		kb, err := val.keyring()
+		s.Require().NoError(err)
+		val0ClientCtx, err := s.chain.clientContext("tcp://localhost:26657", &kb, "val", val.keyInfo.GetAddress())
+		s.Require().NoError(err)
+		auctionQueryClient := types.NewQueryClient(val0ClientCtx)
+
+		s.T().Logf("create governance proposal to update some token prices")
+		orch := s.chain.orchestrators[0]
+		orchClientCtx, err := s.chain.clientContext("tcp://localhost:26657", orch.keyring, "orch", orch.keyInfo.GetAddress())
+		s.Require().NoError(err)
+
+		proposal := types.SetTokenPricesProposal{
+			Title:       "initial token price submission",
+			Description: "our first token prices",
+			TokenPrices: []*types.ProposedTokenPrice{
+				{
+					Denom:    "gravity0x3506424f91fd33084466f402d5d97f05f8e3b4af",
+					UsdPrice: sdk.MustNewDecFromStr("1.0"),
+				},
+				{
+					Denom:    "gravity0x5a98fcbea516cf06857215779fd812ca3bef1b32",
+					UsdPrice: sdk.MustNewDecFromStr("0.25"),
+				},
+				{
+					Denom:    types.UsommDenom,
+					UsdPrice: sdk.MustNewDecFromStr("0.5"),
+				},
+			},
+		}
+
+		proposalMsg, err := govtypes.NewMsgSubmitProposal(
+			&proposal,
+			sdk.Coins{
+				{
+					Denom:  testDenom,
+					Amount: stakeAmount.Quo(sdk.NewInt(2)),
+				},
+			},
+			orch.keyInfo.GetAddress(),
+		)
+		s.Require().NoError(err, "unable to create governance proposal")
+
+		s.T().Log("submit proposal")
+		submitProposalResponse, err := s.chain.sendMsgs(*orchClientCtx, proposalMsg)
+		s.Require().NoError(err)
+		s.Require().Zero(submitProposalResponse.Code, "raw log: %s", submitProposalResponse.RawLog)
+
+		s.T().Log("check proposal was submitted correctly")
+		govQueryClient := govtypes.NewQueryClient(orchClientCtx)
+		proposalsQueryResponse, err := govQueryClient.Proposals(context.Background(), &govtypes.QueryProposalsRequest{})
+		s.Require().NoError(err)
+		s.Require().NotEmpty(proposalsQueryResponse.Proposals)
+		s.Require().Equal(uint64(1), proposalsQueryResponse.Proposals[0].ProposalId, "not proposal id 1")
```

Comment:
The main reason I did it this way is bc the cork and scheduled cork integ tests also do hard coded values. Don't mind updating these values here but theres also potentially some pitfalls of assuming that the most recent proposal doesnt depend on some previous proposal in the genesis state or visa versa. 

E.g. Lets say cellar fees makes some sort of proposal in the test genesis state one day that affects auctions but has to occur after a price update. If we dont remember to specify/update the tests here to make sure the ordering/IDs are correct we could potentially miss a regression vs if we had a hardcoded a value here this would fail loudly and require a reevaluation of the test or logic flows.

Reply:
It's probably fine to leave it to the test breaker to fix it